src/workspace/workspace.component.tsx

```diff
@@ -76,20 +84,23 @@ export default function Workspace(props: any) {
           setSelected={setSelectedTab}
           removeTab={removeTab}
         >
-          {openTabs.map((tab, i) => (
-            <Panel
-              key={i}
-              title={tab.name}
-              style={selectedTab === i ? {} : { display: "none" }}
-            >
-              <tab.component
-                {...tab.props}
-                entryStarted={() => setWorkBegan(i)}
-                entrySubmitted={() => setWorkEnded(i)}
-                entryCancelled={() => setWorkEnded(i)}
-              />
-            </Panel>
-          ))}
+          {openTabs.map((tab, i) => {
+            return (
+              <Panel
+                key={i}
+                title={tab.name}
+                style={selectedTab === i ? {} : { display: "none" }}
+              >
+                <tab.component
+                  {...tab.props}
+                  entryStarted={() => setWorkBegan(i)}
+                  entrySubmitted={() => setWorkEnded(i)}
+                  entryCancelled={() => setWorkEnded(i)}
+                  closeComponent={() => onCloseTabRequest(i, tab)}
```

Comment:
why not use the `entryCancelled` prop for this purpose?

Reply:
I thought about it:
entryCancelled/entrySubmitted is actually to notify the workspace manager, that it is safe to close a certain component hosted in the workspace. The intention might not to unload it from the workspace however. For e.g, after submitting a form, you want to display some information, and have the user close the tab after they are done reading the information. 
However in some case, you want to close(unload) a tab programmatically, and that's why we introduce a new event to close/unload it. 
I hope this makes sense. Let me know if you have any suggestions regarding this.